pkg/cmd/pulumi/pulumi.go

```diff
@@ -722,6 +735,39 @@ func getUpgradeMessage(latest semver.Version, current semver.Version, isDevVersi
 	return msg
 }
 
+// diffVersions compares two semver versions and returns:
+//   - sign: 1 if v1 > v2, -1 if v1 < v2, 0 if they are equal
+//   - majorDiff: the absolute difference in major versions
+//   - minorDiff: the absolute difference in minor versions, or 0 if the major versions differ
+func diffVersions(v1 semver.Version, v2 semver.Version) (sign int, majorDiff uint64, minorDiff uint64) {
+	switch {
+	case v1.Equals(v2):
+		return 0, 0, 0
+	case v1.GT(v2):
+		sign = 1
+	default:
+		sign = -1
+	}
```

Comment:
Maybe make this an enum for readability? Or just not return the sign at all, and use `if v1.LT(v2)` in the condition above.

Reply:
Good shout -- removed the sign and will just check `LT`